head_bench_test.go

```diff
@@ -56,17 +56,56 @@ func BenchmarkHeadPostingForMatchers(b *testing.B) {
 	testutil.Ok(b, err)
 	defer h.Close()
 
-	// TODO: vary number of series
-	for i := 0; i < 1000000; i++ {
-		h.getOrCreate(uint64(i), labels.FromStrings("a", strconv.Itoa(i)))
+	var hash uint64
+	for n := 0; n < 10; n++ {
+		for i := 0; i < 100000; i++ {
+			h.getOrCreate(hash, labels.FromStrings("i", strconv.Itoa(i), "n", strconv.Itoa(i), "j", "foo"))
+			hash++
+			h.getOrCreate(hash, labels.FromStrings("i", strconv.Itoa(i), "n", strconv.Itoa(i), "j", "bar"))
```

Comment:
Why is this needed?

Reply:
So there's some series that won't be matched.

Reply:
yep I understand this part, but why is it needed? 
I would put a comment as well.

Reply:
Done